src/gfx/egui/src/input.rs

```diff
@@ -0,0 +1,267 @@
+        SDL_KeyCode_SDLK_KP_ENTER => Ok(egui::Key::Enter),
+        SDL_KeyCode_SDLK_KP_1 => Ok(egui::Key::Num1),
+        SDL_KeyCode_SDLK_KP_2 => Ok(egui::Key::Num2),
+        SDL_KeyCode_SDLK_KP_3 => Ok(egui::Key::Num3),
+        SDL_KeyCode_SDLK_KP_4 => Ok(egui::Key::Num4),
+        SDL_KeyCode_SDLK_KP_5 => Ok(egui::Key::Num5),
+        SDL_KeyCode_SDLK_KP_6 => Ok(egui::Key::Num6),
+        SDL_KeyCode_SDLK_KP_7 => Ok(egui::Key::Num7),
+        SDL_KeyCode_SDLK_KP_8 => Ok(egui::Key::Num8),
+        SDL_KeyCode_SDLK_KP_9 => Ok(egui::Key::Num9),
+        SDL_KeyCode_SDLK_KP_0 => Ok(egui::Key::Num0),
+        _ => Err(()),
+    }
+}
+
+pub fn to_egui_cursor_icon(i: CursorIcon) -> egui_CursorIcon {
+    match i {
+        CursorIcon::Default => egui_CursorIcon_Default,
+        CursorIcon::None => egui_CursorIcon_None,
+        CursorIcon::ContextMenu => egui_CursorIcon_ContextMenu,
+        CursorIcon::Help => egui_CursorIcon_Help,
+        CursorIcon::PointingHand => egui_CursorIcon_PointingHand,
+        CursorIcon::Progress => egui_CursorIcon_Progress,
+        CursorIcon::Wait => egui_CursorIcon_Wait,
+        CursorIcon::Cell => egui_CursorIcon_Cell,
+        CursorIcon::Crosshair => egui_CursorIcon_Crosshair,
+        CursorIcon::Text => egui_CursorIcon_Text,
+        CursorIcon::VerticalText => egui_CursorIcon_VerticalText,
+        CursorIcon::Alias => egui_CursorIcon_Alias,
+        CursorIcon::Copy => egui_CursorIcon_Copy,
+        CursorIcon::Move => egui_CursorIcon_Move,
+        CursorIcon::NoDrop => egui_CursorIcon_NoDrop,
+        CursorIcon::NotAllowed => egui_CursorIcon_NotAllowed,
+        CursorIcon::Grab => egui_CursorIcon_Grab,
+        CursorIcon::Grabbing => egui_CursorIcon_Grabbing,
+        CursorIcon::AllScroll => egui_CursorIcon_AllScroll,
+        CursorIcon::ResizeHorizontal => egui_CursorIcon_ResizeHorizontal,
+        CursorIcon::ResizeNeSw => egui_CursorIcon_ResizeNeSw,
+        CursorIcon::ResizeNwSe => egui_CursorIcon_ResizeNwSe,
+        CursorIcon::ResizeVertical => egui_CursorIcon_ResizeVertical,
+        CursorIcon::ResizeEast => egui_CursorIcon_ResizeEast,
+        CursorIcon::ResizeSouthEast => egui_CursorIcon_ResizeSouthEast,
+        CursorIcon::ResizeSouth => egui_CursorIcon_ResizeSouth,
+        CursorIcon::ResizeSouthWest => egui_CursorIcon_ResizeSouthWest,
+        CursorIcon::ResizeWest => egui_CursorIcon_ResizeWest,
+        CursorIcon::ResizeNorthWest => egui_CursorIcon_ResizeNorthWest,
+        CursorIcon::ResizeNorth => egui_CursorIcon_ResizeNorth,
+        CursorIcon::ResizeNorthEast => egui_CursorIcon_ResizeNorthEast,
+        CursorIcon::ResizeColumn => egui_CursorIcon_ResizeColumn,
+        CursorIcon::ResizeRow => egui_CursorIcon_ResizeRow,
+        CursorIcon::ZoomIn => egui_CursorIcon_ZoomIn,
+        CursorIcon::ZoomOut => egui_CursorIcon_ZoomOut,
+    }
+}
+
+pub fn translate_raw_input(
+    input: &egui_Input,
+) -> Result<egui::RawInput, Box<dyn std::error::Error>> {
```

Comment:
```suggestion
) -> Result<egui::RawInput, &str> {
```

Reply:
I changed it to 
```
pub enum TranslationError {
    Text(std::str::Utf8Error),
}
```
you should be able to silently ignore that one in the UI shard though, maybe just a log